### PR TITLE
Make CI error on inductor fallback when decomp is available

### PR DESCRIPTION
**Pull request by wconstab**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #99473
* #99659

Fixes #99446 

Remove the warning, as that annoyed end-users who don't know what to do about it.

Instead, try to hold the line by preventing any decomp from being added without making
the corresponding change to inductor's fallbacks.

Note: we probably still need to better document how to update inductor's decomps,
for now it's pretty much "go ask the inductor team for advice"

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/lowering.py [hunk 1]
```diff
@@ -1085,10 +1087,19 @@ def make_fallback(kernel, layout_constraint=None, warn=True):
...
+        log.error(
+            "make_fallback(%s): a decomposition exists, we should switch to it."
+            " To fix this error, either add a decomposition to core_aten_decompositions (preferred)"
+            " or inductor_decompositions, and delete the corresponding `make_fallback` line."
+            " Get help from the inductor team if unsure, don't pick arbitrarily to unblock yourself.",
+            kernel,
         )
+        # Make sure the process exits and CI flags an error, raising
+        # an exception could allow a graph-break and silently pass CI in some configs
+        sys.exit(1)
```
Comment (ezyang) [inline torch/_inductor/lowering.py:1102]: This seems cure is worse than disease. Why don't we raise an exception and write off the suppress error configs?

**Review by ezyang — CHANGES_REQUESTED** [review]

**Review by ngimel — APPROVED** [review]

File: torch/_inductor/lowering.py (hunk 1 above)
Comment (wconstab) [reply torch/_inductor/lowering.py:1102]: I mean, I kinda knew you'd say that. But on balance, this way will stop fallbacks from getting checked in broken, and the only downside that I'm cognizant of is that it robs you of getting notified of further-down-the-stack failures on CI if it exits early.  I think it's slightly better to be sure not to allow bad things to land than to make devs rerun CI more times.  But ...

Comment (ezyang) [reply torch/_inductor/lowering.py:1102]: No, I don't understand your response. What does `sys.exit(0)` give you that `raise AssertionError()` wouldn't?

Comment (wconstab) [reply torch/_inductor/lowering.py:1102]: I am worried that dynamo will swallow exceptions sometimes in some tests in its attempt to gracefully fall back to eager. However, I hope there is some type of internal assertion we can raise that will reliably raise to userland, and if so i'm happy to use that instead.

Comment (jansel) [reply torch/_inductor/lowering.py:1102]: Perhaps we should do
```
if TEST_WITH_DYNAMO:
   sys.exit(1)
else:
   raise ....
```

Under `TEST_WITH_DYNAMO` raising will just fall back silently to eager and skip the real test.


Another option would be to dynamically switch `suppress_errors=False` and raise.

**Review by jansel — APPROVED** [review]

**Comment by wconstab** [comment]
@ngimel please take another look as I have taken the liberty of bypassing one decomp that i found to cause test failures in inductor when using.  If you're ok with this, I can open another issue to track the bad decomp.

**Review by ngimel — APPROVED** [review]
not using `exponential` decomp is fine, but why do you need separate DENY_LIST for it? Can you remove it from extra random decomps and make a fallback instead?

**Comment by wconstab** [comment]
> not using exponential decomp is fine, but why do you need separate DENY_LIST for it? Can you remove it from extra random decomps and make a fallback instead?

Hm,  yes, why did I not do that. I guess i can make an explicit fallback with warn=False for that one.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**